src/foundry/client-esm/audio/timeout.d.mts

```diff
@@ -0,0 +1,82 @@
+/**
+ * A special error class used for cancellation.
+ * @privateRemarks Yes this is a straight up empty class extension by Foundry used exclusively internally
+ */
+declare class AudioTimeoutCancellation extends Error {}
+
+declare namespace AudioTimeout {
+  interface Options<Return> {
+    context?: AudioContext | undefined;
+
+    callback?: (() => Return) | undefined;
+  }
+}
+
+/**
+ * A framework for scheduled audio events with more precise and synchronized timing than using window.setTimeout.
+ * This approach creates an empty audio buffer of the desired duration played using the shared game audio context.
+ * The onended event of the AudioBufferSourceNode provides a very precise way to synchronize audio events.
+ * For audio timing, this is preferable because it avoids numerous issues with window.setTimeout.
+ *
+ * @example Using a callback function
+ * ```js
+ * function playForDuration(sound, duration) {
+ *   sound.play();
+ *   const wait = new AudioTimeout(duration, {callback: () => sound.stop()})
+ * }
+ * ```
+ *
+ * @example Using an awaited Promise
+ * ```js
+ * async function playForDuration(sound, duration) {
+ *   sound.play();
+ *   const timeout = new AudioTimeout(delay);
+ *   await timeout.complete;
+ *   sound.stop();
+ * }
+ * ```
+ *
+ * @example Using the wait helper
+ * ```js
+ * async function playForDuration(sound, duration) {
+ *   sound.play();
+ *   await AudioTimeout.wait(duration);
+ *   sound.stop();
+ * }
+ * ```
+ */
+declare class AudioTimeout<CallbackReturn = void> {
```

Comment:
I'd write `unknown` instead of `void` here. The semantics of `void` assume it's a throwaway value which tbf it probably is but if you check `complete` and see `Promise<unknown>` it indicates there's something as opposed to `Promise<void>` which doesn't.